packages/viewer/src/services.ts

```diff
@@ -0,0 +1,178 @@
+import Parser from '@wiris/mathtype-html-integration-devkit/src/parser';
+
+enum MethodType {
+  Post = "POST",
+  Get = "GET",
+}
+
+/**
+ * Thrown when a service returns a JSON with a non-ok status value in its JSON body
+ */
+export class StatusError extends Error {
+  constructor(message) {
+    super(message);
+    Object.setPrototypeOf(this, StatusError.prototype);
+  }
+}
+
+/**
+ * Helper function to process responses from the editor services.
+ * These usually come wrapped in a JSON with a status field that can be either "ok" or "warning".
+ * If status is "ok", return the result value along it. Otherwise, throw a StatusError.
+ * @param {Promise<Response>} response - The response given by the service.
+ * @returns {Promise<any>} The unwrapped result of the response, if valid.
+ * @throws {StatusError} Service responded with a non-ok status.
+ */
+export async function processJsonResponse(response: Promise<Response>): Promise<any> {
+  try {
+    const { status, result } = await (await response).json();
+
+    if (status !== 'ok') {
+      throw new StatusError('Service responded with a non-ok status');
+    }
+
+    return result;
+  } catch(e) {
+    // TODO manage network and status non-ok errors
+    throw e;
+  }
+}
+
+/**
+ * Calls the endpoint servicename and returns its response.
+ * @param {object} query - Object of parameters to pass as the body request or search parameters.
+ * @param {string} serviceName - Name of the service to be called.
+ * @param {string} serverURL - Url of the server where we want to call the service.
+ * @returns {Promise<Response>} The request response.
+ */
+export async function callService(query: object, serviceName: string, method: MethodType, serverURL: string) : Promise<any> {
+  try {
+    const url = new URL(serviceName, serverURL);
```

Comment:
PHP services need to add the .php extension at the end of the service name. 

If we want to allow users to change the backend could be necessary to add a new property to set the backend technology and add the corresponding suffix on each service call.

Reply:
Good catch! Fixed